zopkio/test_runner.py

```diff
@@ -255,21 +250,39 @@ def _execute_parallel_tests(self, config, failure_handler, naarad_obj, tests):
       else:
         logger.debug("Executing tests: {0}".format([test.name for test in tests]))
         def run_test_command(test):
-          try:
```

Comment:
this function has gotten too long, let's think about how to refactor into smaller parts